### DISCUSSION TITLE
How to push a podman image to docker.io

### DISCUSSION THREAD

**Discussion by Franky45** [containers/podman#22118] [Q&A] [answered]
Hi,
iam using ubuntu 22.04 lts podman4.6.2 version.
I have built an image and i have created a docker registry.
can we push podman images to docker.io?
if possible how to push, provide commands.
i have faced below issue?
when iam trying to push the the locally build podman image to docker registry it is giving error?
how to resolve it?
![Screenshot from 2024-03-21 19-31-09](https://github.com/containers/podman/assets/137598217/8badbc59-cec0-4cb6-b25f-b96d82f51127)
this is the error iam facing
kindly provide solution for this
thanks

**Comment by vrothberg** [comment]
Thanks for reaching out!

Yes, podman can push to _any_ container registry including Docker Hub.  The error suggests that you don't have permissions to push to docker.io/franky2215/.  Did you do a `podman login docker.io` with your users and password?

**Comment by Franky45** [comment]
Hi vrothberg,
Can you send full command podman login docker.io with your users and password or is this the full command

**Reply by vrothberg** [reply]
`podman login docker.io` will prompt you for both user and password.  But you can also provide the credentials directly on the command line via `--username ... --password ...`.  More information on `podman login --help` or `man podman-login`.

**Comment by Franky45** [comment]
It is showing invalid username/password but with same credentials iam able to login using docker login

**Reply by vrothberg** [reply]
Are you behind some proxy?  Any other errors you encounter during login?

**Comment by Franky45** [comment]
Yes I was connected through office wifi which has some firewall iam not sure this creates login issue

**Reply by vrothberg** [reply]
That may be it.  I suggest retrying in another network if possible.

**Reply by Franky45** [reply]
yes i tried by changing the network but the issue remain same

**Reply by vrothberg** [reply]
Can you rerun with `podman --log-level=debug login docker.io` and share the logs?  Please make sure to scan the logs for any private/sensitive user information and remove that before sharing.

**Reply by Franky45** [reply]
DEBU[0000] [graphdriver] trying provided driver "overlay" 
DEBU[0000] Cached value indicated that overlay is supported 
DEBU[0000] Cached value indicated that overlay is supported 
DEBU[0000] Cached value indicated that metacopy is not being used 
DEBU[0000] Cached value indicated that native-diff is usable 
DEBU[0000] backingFs=extfs, projectQuotaSupported=false, useNativeDiff=true, usingMetacopy=false 
DEBU[0000] Initializing event backend journald          
DEBU[0000] Configured OCI runtime runsc initialization failed: no valid executable found for OCI runtime runsc: invalid argument 
DEBU[0000] Configured OCI runtime youki initialization failed: no valid executable found for OCI runtime youki: invalid argument 
DEBU[0000] Configured OCI runtime krun initialization failed: no valid executable found for OCI runtime krun: invalid argument 
DEBU[0000] Configured OCI runtime ocijail initialization failed: no valid executable found for OCI runtime ocijail: invalid argument 
DEBU[0000] Configured OCI runtime crun-wasm initialization failed: no valid executable found for OCI runtime crun-wasm: invalid argument 
DEBU[0000] Configured OCI runtime runj initialization failed: no valid executable found for OCI runtime runj: invalid argument 
DEBU[0000] Configured OCI runtime kata initialization failed: no valid executable found for OCI runtime kata: invalid argument 
DEBU[0000] Using OCI runtime "/usr/bin/crun"            
INFO[0000] Setting parallel job count to 13             
DEBU[0000] Loading registries configuration "/etc/containers/registries.conf" 
DEBU[0000] Loading registries configuration "/etc/containers/registries.conf.d/000-shortnames.conf" 
DEBU[0000] Loading registries configuration "/etc/containers/registries.conf.d/shortnames.conf" 
DEBU[0000] No credentials matching docker.io found in /run/user/1000/containers/auth.json 
DEBU[0000] No credentials matching docker.io found in /home/mahalakshmi/.config/containers/auth.json 
DEBU[0000] No credentials matching docker.io found in /home/mahalakshmi/.docker/config.json 
DEBU[0000] No credentials matching docker.io found in /home/mahalakshmi/.dockercfg 
DEBU[0000] No credentials for docker.io found           
Username: franky2215
Password: 
DEBU[0025] Looking for TLS certificates and private keys in /etc/docker/certs.d/docker.io 
DEBU[0025] GET https://registry-1.docker.io/v2/         
DEBU[0026] Ping https://registry-1.docker.io/v2/ status 401 
DEBU[0026] GET https://auth.docker.io/token?account=franky2215&service=registry.docker.io 
DEBU[0027] error logging into "docker.io": unable to retrieve auth token: invalid username/password: unauthorized: incorrect username or password 
Error: logging into "docker.io": invalid username/password
DEBU[0027] Shutting down engines
these are the logs that are generated

**Reply by vrothberg** [reply, marked as answer]
Thank you for sharing!  The logs indicate that the credentials are incorrect:
> error logging into "docker.io": unable to retrieve auth token: invalid username/password: unauthorized: incorrect username or password

I am not sure how to assist remotely.  To check whether the right credentials are used, could you try running it with Docker?

**Reply by Franky45** [reply]
Authenticating with existing credentials...
WARNING! Your password will be stored unencrypted in /root/.docker/config.json.
Configure a credential helper to remove this warning. See
https://docs.docker.com/engine/reference/commandline/login/#credentials-store

Login Succeeded

yes,  tried with docker login is succeeded

**Comment by Franky45** [comment]
Ok, but through docker login logging is getting successful

**Comment by Franky45** [comment]
the credentials are incorrect:

**Reply by afkzoro** [reply]
Hey @Franky45 did it work eventually?

**Comment by Franky45** [comment]
Yes
